#!/usr/bin/env python
PACKAGE = "spc_uav_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("mass",   	double_t, 0, "Mass",		  		              		  1.77, 0.00,  7.00)
gen.add("J_xx", 	double_t, 0, "Inertia Matrix xx", 	        			  0.0347563, 0.00,  0.50)
gen.add("J_yy", 	double_t, 0, "Inertia Matrix yy", 	        			  0.0458929, 0.00,  0.50)
gen.add("J_zz", 	double_t, 0, "Inertia Matrix zz", 	        			  0.0977, 0.00,  0.50)
gen.add("K_p_x", 	double_t, 0, "Position Gain - x", 	        			  6.00, 0.00,  10.00)
gen.add("K_p_y", 	double_t, 0, "Position Gain - y", 	        			  6.00, 0.00,  10.00)
gen.add("K_p_z", 	double_t, 0, "Position Gain - z", 	        			  6.00, 0.00,  10.00)
gen.add("K_v_x", 	double_t, 0, "Velocity Gain - x", 	        			  4.70, 0.00,  10.00)
gen.add("K_v_y", 	double_t, 0, "Velocity Gain - y", 	        			  4.70, 0.00,  10.00)
gen.add("K_v_z", 	double_t, 0, "Velocity Gain - z", 	        			  4.70, 0.00,  10.00)
gen.add("K_R_x", 	double_t, 0, "Attitude Gain - x", 	        			  3.00, 0.00,  10.00)
gen.add("K_R_y", 	double_t, 0, "Attitude Gain - y", 	        			  3.00, 0.00,  10.00)
gen.add("K_R_z", 	double_t, 0, "Attitude Gain - z", 				          3.00, 0.00,  10.00)
gen.add("K_w_x", 	double_t, 0, "Angular Rate Gain - x", 				      0.5, 0.00,  10.00)
gen.add("K_w_y", 	double_t, 0, "Angular Rate Gain - y", 				      0.5, 0.00,  10.00)
gen.add("K_w_z", 	double_t, 0, "Angular Rate Gain - z", 				      0.5, 0.00,  10.00)
gen.add("K_w_x_px4",double_t, 0, "PX4 Onboard Angular Rate Gain - x", 		  0.2, 0.01,  10.00)
gen.add("K_w_y_px4",double_t, 0, "PX4 Onboard Angular Rate Gain - y", 		  0.2, 0.01,  10.00)
gen.add("K_w_z_px4",double_t, 0, "PX4 Onboard Angular Rate Gain - z", 		  0.3, 0.01,  10.00)
gen.add("K_0_t", 	double_t, 0, "Disturbance Observer Translational Gain 0", 0.0, 0.00,  50.00)
gen.add("K_0_r", 	double_t, 0, "Disturbance Observer Rotational Gain 0", 	  0.0, 0.00,  50.00)
gen.add("K_1_t", 	double_t, 0, "Disturbance Observer Translational Gain 1", 0.0, 0.00,  50.00)
gen.add("K_1_r", 	double_t, 0, "Disturbance Observer Rotational Gain 1", 	  0.0, 0.00,  50.00)
gen.add("DO_ON", 	double_t, 0, "Disturbance Observer Switch", 			  0.0, 0.00,  1.00)
exit(gen.generate(PACKAGE, "spc_uav_control", "SE3_Control_Gains"))
